### PR TITLE
[GNA] Added SW_FP32 mode w/o SF for BasicLSTM

### PR DESCRIPTION
**Pull request by a-noskov**
### Details:
 - Disabled TI3AlignedPropagateForward and TI1AlignedPropagateForward were moved to new infrastructure w/o using old version IR
 - Added GNA_SW_FP32 mode for Basic_LSTM, so the weights (Input, MatMul and Hidden constants) are much smaller, than in Basic_LSTM_S test

### Tickets:
 - *31901*

### CODE REVIEW

File: src/tests/functional/plugin/gna/shared_tests_instances/subgraph_tests/basic_lstm_sw_fp32.cpp [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    Run();
+};
+
+const std::vector<InferenceEngine::Precision> netPrecisions = {
+        InferenceEngine::Precision::FP32,
+        InferenceEngine::Precision::FP16
+};
+
+std::map<std::string, std::string> config_no_sf = {
+    {"GNA_COMPACT_MODE", "NO"},
```
Comment (mryzhov) [inline src/tests/functional/plugin/gna/shared_tests_instances/subgraph_tests/basic_lstm_sw_fp32.cpp:116]: We don't need to test it with disabled compact mode, please remove it

File: src/tests/functional/plugin/gna/shared_tests_instances/subgraph_tests/basic_lstm_sw_fp32.cpp [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    {"GNA_DEVICE_MODE", "GNA_SW_FP32"}
```
Comment (mryzhov) [inline src/tests/functional/plugin/gna/shared_tests_instances/subgraph_tests/basic_lstm_sw_fp32.cpp:117]: We need to test the both GNA_SW_EXACT and GNA_SW_FP32 modes every time if it is possible

**Review by mryzhov — CHANGES_REQUESTED** [review]
What is the difference of the test with the existed basic_lstm test?

File: src/tests/functional/plugin/gna/shared_tests_instances/subgraph_tests/basic_lstm_sw_fp32.cpp [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+#include "shared_test_classes/subgraph/basic_lstm.hpp"
+#include <transformations/op_conversions/lstm_cell_decomposition.hpp>
+
+using namespace SubgraphTestsDefinitions;
+
+namespace {
+
+class Basic_LSTM_SW_FP32 : public Basic_LSTM_S {
+public:
+    std::shared_ptr<ngraph::Function> GetNetwork_sw_fp32(size_t thirdDimOut,
```
Comment (nkogteva) [inline src/tests/functional/plugin/gna/shared_tests_instances/subgraph_tests/basic_lstm_sw_fp32.cpp:14]: Can we reuse SubgraphTestsDefinitions::Basic_LSTM_S::GetNetwork?

**Review by mryzhov — CHANGES_REQUESTED** [review]
Please do not merge until we known why SW_FP32 and SW_EXACT work differently for the same network

File: src/tests/functional/shared_test_classes/src/subgraph/basic_lstm.cpp [hunk 4]
```diff
@@ -145,12 +145,18 @@ void Basic_LSTM_S::Run() {
     Infer();
 
     const auto& actualOutputs = GetOutputs();
-    auto referenceOutputs = CalculateRefs();
+    auto gna_mode = configuration.find("GNA_DEVICE_MODE");
```
Comment (nkogteva) [inline src/tests/functional/shared_test_classes/src/subgraph/basic_lstm.cpp:148]: It is common place for all plugins. We should not introduce here GNA specific.

**Review by mryzhov — CHANGES_REQUESTED** [review]
Please remove gna specific code from the common test and replace hardcoded input range with test parameters as we discussed on previous review

File: src/tests/functional/plugin/gna/shared_tests_instances/subgraph_tests/basic_lstm.cpp [hunk 5]
```diff
@@ -31,6 +35,8 @@ size_t small_num_cells = 10;
 
 size_t big_num_cells = 49;
 
+float weights_range = 0.02f;
```
Comment (mryzhov) [inline src/tests/functional/plugin/gna/shared_tests_instances/subgraph_tests/basic_lstm.cpp:38]: let's use pair instead of maximum value and hardcoded "0"

**Review by mryzhov — APPROVED** [review]

**Review by nkogteva — APPROVED** [review]

**Pull request merged**